src/tests/cmocka/test_nested_groups.c

```diff
@@ -237,6 +240,81 @@ static void nested_groups_test_one_group_unique_members(void **state)
                                        expected, N_ELEMENTS(expected));
 }
 
+static void nested_groups_test_one_group_unique_members_one_ignored(void **state)
+{
+    struct nested_groups_test_ctx *test_ctx = NULL;
+    struct sdap_search_base **ignore;
+    struct sysdb_attrs *rootgroup = NULL;
+    struct tevent_req *req = NULL;
+    TALLOC_CTX *req_mem_ctx = NULL;
+    errno_t ret;
+    const char *users[] = { "cn=user1," USER_BASE_DN,
+                            "cn=user2," EXCLUDE_BASE_DN,
+                            NULL };
+    const struct sysdb_attrs *user1_reply[2] = { NULL };
+    const char * expected[] = { "user1" };
+
+
+    test_ctx = talloc_get_type_abort(*state, struct nested_groups_test_ctx);
+
+    /* mock return values */
+    rootgroup = mock_sysdb_group_rfc2307bis(test_ctx, GROUP_BASE_DN, 1000,
+                                            "rootgroup", users);
+
+    /* Set the exclude bases */
+    ignore = talloc_zero_array(test_ctx, struct sdap_search_base *, 3);
+    assert_non_null(ignore);
+
+    ignore[0] = talloc_zero(test_ctx, struct sdap_search_base);
+    assert_non_null(ignore[0]);
+    ignore[0]->basedn = BAD_BASE_DN;
+    ignore[0]->ldb_basedn = ldb_dn_new(test_ctx,
+                                       sysdb_ctx_get_ldb(test_ctx->tctx->sysdb),
+                                       ignore[0]->basedn);
+
```

Comment:
To be one the safe side I'd suggest to add a `assert_non_null(gnore[0]->ldb_basedn)`.